java/test/src/main/java/org/ray/api/test/ReferenceCountingTest.java

```diff
@@ -136,4 +138,20 @@ public void testDependencyRefCounts() {
     ((RayObjectImpl<?>) result).removeLocalReference();
     checkRefCounts(ImmutableMap.of());
   }
+
+  public void testBasicNestedIds() {
+    RayObject<byte[]> inner = Ray.put(new byte[40 * 1024 * 1024]);
+    RayObject<List<RayObject<byte[]>>> outer = Ray.put(Collections.singletonList(inner));
+
+    byte[] innerObjecrIdBytes = inner.getId().getBytes();
+    inner = null;
+    System.gc();
```

Comment:
`System.gc()` is not guaranteed. You can use `((RayObjectImpl<?>) obj).removeLocalReference()` to mock the GC behavior.

Reply:
👌